android/java/org/chromium/chrome/browser/vpn/BraveVpnParentActivity.java

```diff
@@ -0,0 +1,232 @@
+/**
+ * Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package org.chromium.chrome.browser.vpn;
+
+import static com.android.billingclient.api.BillingClient.SkuType.SUBS;
+
+import android.content.Context;
+import android.content.Intent;
+import android.graphics.Paint;
+import android.net.VpnManager;
+import android.os.Bundle;
+import android.util.Pair;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.Button;
+import android.widget.TextView;
+
+import androidx.appcompat.app.ActionBar;
+import androidx.appcompat.widget.Toolbar;
+import androidx.viewpager.widget.ViewPager;
+
+import com.android.billingclient.api.BillingClient;
+import com.android.billingclient.api.BillingClientStateListener;
+import com.android.billingclient.api.BillingResult;
+import com.android.billingclient.api.Purchase;
+import com.android.billingclient.api.PurchasesUpdatedListener;
+import com.google.android.material.tabs.TabLayout;
+
+import org.json.JSONException;
+
+import org.chromium.base.Log;
+import org.chromium.chrome.R;
+import org.chromium.chrome.browser.firstrun.FirstRunFlowSequencer;
+import org.chromium.chrome.browser.init.AsyncInitializationActivity;
+import org.chromium.chrome.browser.vpn.BraveVpnConfirmDialogFragment;
+import org.chromium.chrome.browser.vpn.BraveVpnNativeWorker;
+import org.chromium.chrome.browser.vpn.BraveVpnObserver;
+import org.chromium.chrome.browser.vpn.BraveVpnPlanPagerAdapter;
+import org.chromium.chrome.browser.vpn.BraveVpnPrefUtils;
+import org.chromium.chrome.browser.vpn.BraveVpnProfileUtils;
+import org.chromium.chrome.browser.vpn.BraveVpnUtils;
+import org.chromium.ui.widget.Toast;
+
+import java.util.List;
+import java.util.TimeZone;
+
```

Comment:
classes `BraveActivity` and `BraveVpnParentActivity` have similar methods (though not fully equal):
- `verifySubscription`;
- `onVerifyPurchaseToken`;

and they both work with vpn prefs. Is it possible to avoid duplication?